File: Apps/RoomManagement/MobileApp/src/types/room.ts
export interface RoomType {
  id: number;
  name: string;
  code: string;
  description: string;
  defaultCapacity: number;
  defaultPrice: number;
  icon: string;
  standardAmenities: string[];
  imageUrl: string;
}

export interface Room {
  id: number;
  hotelId: number;
  hotelName: string;
  roomTypeId: number;
  roomTypeName?: string;
  roomNumber: string;
  floor: number;
  view: string;
  capacity: number;
  extraBedCapacity: number;
  basePrice: number;
  isAvailable: boolean;
  isClean: boolean;
  status: string;
  amenities: string[];
  description: string;
}

export interface RoomInventory {
  id: number;
  roomId: number;
  date: string;
  isAvailable: boolean;
  price: number;
  availableCount: number;
  bookedCount: number;
  maintenanceCount: number;
}

export interface RoomStats {
  totalRooms: number;
  availableRooms: number;
  occupiedRooms: number;
  maintenanceRooms: number;
  cleaningRooms: number;
  byRoomType: { roomType: string; count: number }[];
  averagePrice: number;
  todayRevenue: number;
}

File: Apps/RoomManagement/Backend/RoomManagement.API/Models/Room.cs
namespace RoomManagement.API.Models
{
    public class Room
    {
        public int Id { get; set; }
        public int HotelId { get; set; }
        public string HotelName { get; set; }
        public int RoomTypeId { get; set; }
        public string RoomNumber { get; set; }
        public int Floor { get; set; }
        public string View { get; set; } // Sea, City, Mountain, Garden
        public int Capacity { get; set; }
        public int ExtraBedCapacity { get; set; }
        public decimal BasePrice { get; set; }
        public bool IsAvailable { get; set; }
        public bool IsClean { get; set; }
        public string Status { get; set; } // Available, Occupied, Maintenance, Cleaning
        public List<string> Amenities { get; set; }
        public string Description { get; set; }
    }
}